lib/system/sysio.nim

```diff
@@ -270,12 +270,33 @@ const
     # we always use binary here as for Nim the OS line ending
     # should not be translated.
 
+when defined(posix) and not defined(nimscript):
```

Comment:
Could we import the necessary types/procs from the ``posix`` module here? 

But I guess you want to avoid that dependency.